pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,283 @@
+
+        if not inference_args:
+            inference_args = self._set_default_inference_args()
+
+        if not self.cached_model:
+            self.cached_model = self.create_model()
+
+        if minibatch_size:
+            with self.cached_model:
+                minibatches = {
+                    self.shared_vars['model_input']: pm.Minibatch(X, batch_size=minibatch_size),
+                    self.shared_vars['model_output']: pm.Minibatch(y, batch_size=minibatch_size),
+                }
+
+                inference_args['more_replacements'] = minibatches
+        else:
+            self._set_shared_vars({'model_input': X, 'model_output': y})
+
+        self._inference(inference_type, inference_args)
+
+        return self
+
+    def predict_proba(self, X):
+        """
+        Predicts the probabilities that the data points belong to each category.
+
+        Given a new data point :math:`\\vec{x}`, we want to estimate the probability that
+        it belongs to a category :math:`c`. Following the notations in [1], the probability
+        reads:
+
+        .. math::
+
+            P(y=c|\\vec{x}, \\mathcal{D}) = P(y=c|\\mathcal{D}) \\prod_{j=1}^{n_{dims}} \\
+                                            P(x_j|y=c, \\mathcal{D})
+
+        We previously used the data :math:`\\mathcal{D}` to estimate the
+        distribution of the parameters :math:`\\vec{\\mu}`, :math:`\\vec{\\pi}`
+        and :math:`\\vec{\\sigma}`. To compute the above probability, we need
+        to integrate over the values of these parameters:
+
+        .. math::
+
+            P(y=c|\\vec{x}, \\mathcal{D}) = \\left[\\int Cat(y=c|\\vec{\\pi})P(\\vec{\\pi}|\\
+                                            \\mathcal{D})\\mathrm{d}\\vec{\\pi}\\right]
+                                            \\int P(\\vec{x}|\\vec{\\mu}, \\vec{\\sigma})\\
+                                            P(\\vec{\\mu}|\\mathcal{D})\\
+                                            P(\\vec{\\sigma}|\\mathcal{D})\\
+                                            \\mathrm{d}\\vec{\\mu}\\mathrm{d}\\vec{\\sigma}
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the points
+            for which we want to predict the class
+
+        Returns
+        -------
+        A numpy array of shape [num_training_samples, num_cats] that contains the probabilities
+        that each sample belong to each category.
+
+        References
+        ----------
+        .. [1] Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+        """
+
+        if self.trace is None:
+            raise PyMC3ModelsError("Run fit() on the model before predict()")
+
+        predictions = np.array([])
+        for x in X:
+            prob_per_sample = scipy.stats.norm(self.trace['mu'], self.trace['sigma']).pdf(x)
+            prob_per_feature = [np.sum(prob_per_sample[:, :, i], axis=0)/len(self.trace['mu'])
```

Comment:
Since this line is too long, could you format it like this:
```
prob_per_feature =  [
      np.sum(prob_per_sample[:, :, i], axis=0)/len(self.trace['mu']) for i in range(self.num_pred)
]
```

I find that easier to read.